src/login/logind-session-device.c

```diff
@@ -387,14 +407,11 @@ int session_device_new(Session *s, dev_t dev, SessionDevice **out) {
         }
         sd->fd = r;
 
-        LIST_PREPEND(sd_by_device, sd->device->session_devices, sd);
-
         *out = sd;
         return 0;
 
 error:
-        hashmap_remove(s->devices, &sd->dev);
-        free(sd->node);
+        session_device_deinit(sd);
         free(sd);
```

Comment:
We usually have a function `blah_free(blah_t* blah)`, which also frees `blah`. All calls to `session_device_deinit(sd)` are followed by `free(sd)`. It'd be nicer to follow the usual pattern here.